File: Classes/Endereco.java
package Classes;

/* Classe responsável ṕor armazenar um endereço.
 A classe Classes.Endereco é uma classe de apoio, que não será instanciada diretamente, mas sim por meio de uma instância de Classes.Proprietario ou Classes.Imovel.
*/
public class Endereco {
    private String cep;
    private Estados estado;
    private String cidade;
    private String rua;
    private int numero;

    public Endereco(String cep, String estado, String cidade, String rua, int numero) {
        this.rua = rua;
        this.cidade = cidade;
        this.estado = Estados.valueOf(estado);
        this.cep = cep;
        this.numero = numero;
    }

    public String getRua() {
        return rua;
    }

    public String getCidade() {
        return cidade;
    }

    public String getEstado() {
        return estado.getNome();
    }

    public String getCep() {
        return cep;
    }

    public int getNumero() {
        return numero;
    }

    public void setRua(String rua) {
        this.rua = rua;
    }

    public void setCidade(String cidade) {
        this.cidade = cidade;
    }

    public void setEstado(String estado) {
        this.estado = Estados.valueOf(estado);
    }

    public void setCep(String cep) {
        this.cep = cep;
    }

    public void setNumero(int numero) {
        this.numero = numero;
    }

    // Possibilidade de atualização de endereço, para uso da classe Proprietário.
    public void setEndereco(String cep, String estado, String cidade, String rua, int numero) {
        this.cep = cep;
        this.estado = Estados.valueOf(estado);
        this.cidade = cidade;
        this.rua = rua;
        this.numero = numero;
    }

/* TESTE
    public static void main(String[] args) {
        Classes.Endereco endereco = new Classes.Endereco("12345678", "BA", "Salvador", "Rua 1", 1);
        System.out.println(endereco.getEstado());
        endereco.setEstado("SP");
        System.out.println(endereco.getEstado());
    }

 */
}

File: incremental_python/endereco.py
# classe endereço Questão 2

class endereco:

    def __init__(self, cep, estado, cidade, rua, numero):
        self.cep = cep
        self.estado = estado
        self.cidade = cidade
        self.rua = rua
        self.numero = numero

    #gets e sets

    def get_cep(self):
        return self.cep
    
    def set_cep(self, cep):
        self.cep = cep

    def get_estado(self):
        return self.estado
    
    def set_estado(self, estado):
        self.estado = estado

    def get_cidade(self):
        return self.cidade
    
    def set_cidade(self, cidade):
        self.cidade = cidade

    def get_rua(self):
        return self.rua
    
    def set_rua(self, rua):
        self.rua = rua

    def get_numero(self):
        return self.numero
    
    def set_numero(self, numero):
        self.numero = numero

    def equals(self, endereco):
        if self.cep == endereco.get_cep() and self.estado == endereco.get_estado() and self.cidade == endereco.get_cidade() and self.rua == endereco.get_rua() and self.numero == endereco.get_numero():
            return True
        else:
            return False
    
    def __str__(self):
        return "CEP: " + self.cep + "\nEstado: " + self.estado + "\nCidade: " + self.cidade + "\nRua: " + self.rua + "\nNúmero: " + str(self.numero)